frontend/src/components/readsets/ReadsetsTableColumns.tsx

```diff
@@ -0,0 +1,194 @@
+            sorter: true,
+            render: (_, { readset }) => {
+                return readset && readset.metrics && readset.metrics['mean_quality_score'] ? <div> {Number(readset.metrics['mean_quality_score'].value_numeric).toFixed(3)} </div> : ''
+            }
+        },
+        [ReadsetColumnID.BLAST_HIT]: {
+            columnID: ReadsetColumnID.BLAST_HIT,
+            title: "Blast hit",
+            dataIndex: ['readset', 'blast_hit'],
+            sorter: true,
+            render: (_, { readset }) => {
+                return readset && readset.metrics && readset.metrics['1st_hit'] ? <div> {readset.metrics['1st_hit'].value_string} </div> : ''
+            }
+        },
+    }
+}
+
+export enum ReadsetFilterID {
+    ID = ReadsetColumnID.ID,
+    SAMPLE_NAME = ReadsetColumnID.SAMPLE_NAME,
+    RELEASE_STATUS = ReadsetColumnID.RELEASE_STATUS,
+    LIBRARY_TYPE = ReadsetColumnID.LIBRARY_TYPE,
+    INDEX = ReadsetColumnID.INDEX,
+    NUM_READS = ReadsetColumnID.NB_READS,
+    NUM_BASES = ReadsetColumnID.NB_BASES,
+    MEAN_QUALITY_SCORE = ReadsetColumnID.MEAN_QUALITY_SCORE,
+    BLAST_HIT = ReadsetColumnID.BLAST_HIT,
+}
+
+export const READSET_COLUMN_FILTERS: { [key in ReadsetColumnID]: FilterDescription } = {
+    [ReadsetColumnID.ID]: {
+        type: FILTER_TYPE.INPUT_OBJECT_ID,
+        key: UNDEFINED_FILTER_KEY,
+        label: "id",
+    },
+    [ReadsetColumnID.SAMPLE_NAME]: {
+        type: FILTER_TYPE.INPUT_OBJECT_ID,
+        key: UNDEFINED_FILTER_KEY,
+        label: "sample_name",
+    },
+    [ReadsetColumnID.RELEASE_STATUS]: {
+        type: FILTER_TYPE.INPUT_OBJECT_ID,
+        key: UNDEFINED_FILTER_KEY,
+        label: "release_status",
+    },
+    [ReadsetColumnID.LIBRARY_TYPE]: {
+        type: FILTER_TYPE.INPUT_OBJECT_ID,
+        key: UNDEFINED_FILTER_KEY,
+        label: "library_type",
+    },
+    [ReadsetColumnID.INDEX]: {
+        type: FILTER_TYPE.INPUT_OBJECT_ID,
+        key: UNDEFINED_FILTER_KEY,
+        label: "index",
+    },
+    [ReadsetColumnID.NB_READS]: {
+        type: FILTER_TYPE.INPUT_OBJECT_ID,
+        key: UNDEFINED_FILTER_KEY,
+        label: "nb_reads",
+    },
+    [ReadsetColumnID.NB_BASES]: {
+        type: FILTER_TYPE.INPUT_OBJECT_ID,
+        key: UNDEFINED_FILTER_KEY,
+        label: "nb_bases",
+    },
+    [ReadsetColumnID.MEAN_QUALITY_SCORE]: {
+        type: FILTER_TYPE.INPUT_OBJECT_ID,
+        key: UNDEFINED_FILTER_KEY,
+        label: "mean_qualirt_score",
+    },
+    [ReadsetColumnID.BLAST_HIT]: {
+        type: FILTER_TYPE.INPUT_OBJECT_ID,
+        key: UNDEFINED_FILTER_KEY,
+        label: "blast_hit",
+    }
+}
+
+export const READSET_FILTER_KEYS: { [key in ReadsetColumnID]: string } = {
+    [ReadsetColumnID.ID]: 'id',
+    [ReadsetColumnID.SAMPLE_NAME]: 'sample_name',
+    [ReadsetColumnID.RELEASE_STATUS]: 'release_status',
+    [ReadsetColumnID.LIBRARY_TYPE]: 'library_type',
+    [ReadsetColumnID.INDEX]: 'index',
+    [ReadsetColumnID.NB_READS]: 'nb_reads',
```

Comment:
Hmm. To filter by metric value we would need a key here that matches a metric by name, eg 'nb_reads'. I'm not sure how to formulate a key like that. Please check with Ulysse.

Reply:
I'm not sure myself if you can have a single key to do that ... it would be the combination of metrics__name for the type of metrics and metrics__value_numeric. Since we do not really want to do something flexible, you can add an annotation on the readset viewset to expose the nb_reads specifically it should allow you to filter using that key (something like readset_reads)...